test_agent/rust/src/main.rs

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+mod constants;
+
+#[path = "../../../up_client_socket/rust/u_transport_socket.rs"]
+pub mod u_transport_socket;
+ 
+mod utils;
+
+use std::thread;
+
+use crate::constants::*;
+use testagent::SocketTestAgent;
+use u_transport_socket::{UtransportExt, UtransportSocket};
+mod testagent;
+use tokio::runtime::Runtime;
+use std::net::TcpStream as TcpStreamSync;
+
+fn main() {
```

Comment:
You can make this an async main by using an attribute macro so that it cleans up some of the stuff you're doing down below:
```suggestion
#[tokio::main]
fn main() {
```